test/extensions/filters/network/zookeeper_proxy/config_test.cc

```diff
@@ -43,6 +43,91 @@ max_packet_bytes: -1
   EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), EnvoyException);
 }
 
+TEST(ZookeeperFilterConfigTest, UndefinedOpcode) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+latency_threshold_overrides:
+  - opcode: Undefined
+    threshold:
+      nanos: 150000000
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config),
+               ProtobufMessage::UnknownProtoFieldException);
+}
+
+TEST(ZookeeperFilterConfigTest, NegativeLatencyThreshold) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+latency_threshold_overrides:
+  - opcode: Connect
+    threshold:
+      nanos: -150000000
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), EnvoyException);
+}
+
+TEST(ZookeeperFilterConfigTest, TooSmallLatencyThreshold) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+latency_threshold_overrides:
+  - opcode: Multi
+    threshold:
+      nanos: 999999
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), EnvoyException);
+}
+
+TEST(ZookeeperFilterConfigTest, UnsetLatencyThreshold) {
+  const std::string yaml = R"EOF(
+stat_prefix: test_prefix
+latency_threshold_overrides:
+  - opcode: Create
+    threshold:
+  )EOF";
+
+  ZooKeeperProxyProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), EnvoyException);
```

Comment:
Scan for text identifying the error is included in the exception using EXPECT_THROW_WITH_MESSAGE?